src/utils/getTokenPriceInUSD.ts

```diff
@@ -0,0 +1,175 @@
+    {
+      inputs: [],
+      name: "latestRoundData",
+      outputs: [
+        { internalType: "uint80", name: "roundId", type: "uint80" },
+        { internalType: "int256", name: "answer", type: "int256" },
+        { internalType: "uint256", name: "startedAt", type: "uint256" },
+        { internalType: "uint256", name: "updatedAt", type: "uint256" },
+        { internalType: "uint80", name: "answeredInRound", type: "uint80" },
+      ],
+      stateMutability: "view",
+      type: "function",
+    },
+    {
+      inputs: [],
+      name: "version",
+      outputs: [{ internalType: "uint256", name: "", type: "uint256" }],
+      stateMutability: "view",
+      type: "function",
+    },
+  ];
+  // The address of the contract which will provide the price of ETH
+  const feedAddress = tokenAddressToFeedAddress(chainId, tokenAddress);
+
+  if (!feedAddress) {
+    throw new Error("Token not supported");
+  }
+
+  // We create an instance of the contract which we can interact with
+  const priceFeed = new ethers.Contract(
+    feedAddress,
+    aggregatorV3InterfaceABI,
+    provider,
+  );
+  // We get the data from the last round of the contract
+  // Determine how many decimals the price feed has (10**decimals)
+  const [roundData, decimals] = await Promise.all([
+    priceFeed.latestRoundData(),
+    priceFeed.decimals(),
+  ]);
+  // We convert the price to a number and return it
+  return Number(
+    (roundData.answer.toString() / Math.pow(10, Number(decimals))).toFixed(2),
+  );
+};
+
+const tokenAddressToFeedAddress = (chainId: ChainId, tokenAddress: string) => {
+  const currencies = currenciesByNetwork[chainId];
+  const currency = Object.values(currencies).find(
+    (currency) => currency.address === tokenAddress,
+  );
+
+  if (!currency) {
+    throw new Error("Currency not supported");
+  }
+
+  const symbol = currency.symbol as keyof (typeof currenciesByNetwork)[ChainId];
+  const feedsForChain = feedsPerChain[chainId];
+
+  if (!feedsForChain) {
+    throw new Error("Chain not supported");
+  }
+
+  return feedsForChain?.[symbol];
+};
+
+export const getTokenPricesForChain = async (chainId: ChainId) => {
+  const currencies = currenciesByNetwork[chainId];
+  const prices = await Promise.all(
+    Object.values(currencies).map(async (currency: Currency) => {
+      try {
+        const price = await getTokenPriceInUSD(chainId, currency.address);
+        return { ...currency, price };
+      } catch (error) {
+        console.error(error);
+        return { ...currency, price: 0 };
+      }
+    }),
+  );
+
+  return prices.reduce(
+    (acc, result) => {
+      return { ...acc, [result.address]: result };
+    },
+    {} as Record<string, Currency & { price: number }>,
+  );
+};
+
+type CurrencyFeeds = Record<
+  keyof (typeof currenciesByNetwork)[ChainId],
+  string
+>;
+
+const feedsPerChain: Record<ChainId, Partial<CurrencyFeeds>> = {
+  [ChainId.SEPOLIA]: {
+    ETH: "0x694AA1769357215DE4FAC081bf1f309aDC325306",
```

Comment:
when I run the API I get a 429 on the call to this address. 

"message: 'Your app has exceeded its compute units per second capacity. If you have retries enabled, you can safely ignore this message. If not, check out https://docs.alchemy.com/reference/throughput'
"

Reply:
How did you trigger this exactly? Simply by starting up the API? Because I'm not able to reproduce it, but would like to to see if a solution is necessary.

Reply:
Starting the app, but it might've been simply my API key. After merging, please monitor for errors in the Railway logs while testing out the deployed flow